docs/code-generation/generate-hcl.md

```diff
@@ -295,30 +310,74 @@ generate_hcl "file.tf" {
 }
 ```
 
-And if `global.values` is undefined the block is just ignored.
+In this example, if the `global.values` is undefined, the `tm_dynamic` block is ignored during code generation.
+
+
+# Hierarchical Code Generation
+
+Hierarchical code generation in Terramate allows defining code generation at different levels in a project. It provides flexibility and control over the generated code. 
+
+By specifying code generation rules at various levels, you can achieve granular control over different stacks and the resources that are associated to them.
+
+## Defining Code Generation at different levels in a project
```

Comment:
```suggestion
## Defining Code Generation at different levels in a project
```

inconsistencies in upper and lower case - we discussed this the other day. Please follow a clear pattern.